core/src/main/java/org/elasticsearch/index/mapper/AllFieldMapper.java

```diff
@@ -112,7 +111,7 @@ public AllFieldMapper build(BuilderContext context) {
             // the AllFieldMapper ctor in the builder since it is not valid. Here we validate
             // the doc values settings (old and new) are rejected
             Object docValues = node.get("doc_values");
-            if (docValues != null && lenientNodeBooleanValue(docValues)) {
+            if (docValues != null && TypeParsers.nodeBooleanValue(name, "doc_values", docValues, parserContext)) {
```

Comment:
Given that `_all` fields cannot be configured for 6.0+ indices (see #22144), I'd also say we should be lenient here. Thanks for catching this.